07-routing-gossip.md

```diff
@@ -795,9 +799,15 @@ If the incoming message includes `query_option`, the receiver MAY append additio
 
 #### Rationale
 
-A single response might be too large for a single packet, and also a peer can
-store canned results for (say) 1000-block ranges, and simply offer each reply
-which overlaps the ranges of the request.
+A single response might be too large for a single packet, so multiple replies
+may be required.  We want to allow a peer to store canned results for (say)
+1000-block ranges, so replies can exceed the requested range.  However, we
+require that each reply be relevant (overlapping the requested range).
+
+By insisting that replies be in increasing order, the receiver can easily
+determine if replies are done: simply check if `first_blocknum` plus
```

Comment:
I think this doesn't work in all cases...
For example when we split the last requested block between multiple responses:

```
* query: first_blocknum = 10, number_of_blocks = 2
* response 1: first_blocknum = 10, number_of_blocks = 2, short_channel_ids = {10x1x1, 11x0x2}
* response 2: first_blocknum = 11, number_of_blocks = 1, short_channel_ids = {11x1x1}
```

You will consider it done after response 1.

Reply:
No, we assume that a single block's channels are always in the same response.  Otherwise you'd never know when to terminate.

Reply:
Then why do you need the "minus 1" line 791?